turbinia/workers/__init__.py

```diff
@@ -609,14 +629,40 @@ def run_wrapper(self, evidence):
 
     log.debug('Task {0:s} {1:s} awaiting execution'.format(self.name, self.id))
     evidence = evidence_decode(evidence)
+    try:
+      self.result = self.setup(evidence)
+      self.result.update_task_status(self, 'queued')
+    except TurbiniaException as exception:
+      message = (
+          '{0:s} Task setup failed with exception: [{1!s}]'.format(
+              self.name, exception))
+      # Logging explicitly here because the result is in an unknown state
+      trace = traceback.format_exc()
+      log.error(message)
+      log.error(trace)
+      if self.result:
+        if hasattr(exception, 'message'):
+          self.result.set_error(exception.message, traceback.format_exc())
+        else:
+          self.result.set_error(exception.__class__, traceback.format_exc())
+        self.result.status = message
+      else:
+        self.result = TurbiniaTaskResult(
+            base_output_dir=self.base_output_dir, request_id=self.request_id,
+            job_id=self.job_id)
+        self.result.setup(self)
+        self.result.status = message
+        self.result.set_error(message, traceback.format_exc())
+        return self.result.serialize()
```

Comment:
I think we want to actually bring this out of this if/else statement (i.e. keep it on the same line, but just dedent it two spaces) so that we return the result whether or not we had to create a new one, right?

Reply:
Fixed.